definitions/procedures/puppet/disable_puppet.rb

```diff
@@ -0,0 +1,45 @@
+module Procedures::Puppet
+  class DisablePuppet < ForemanMaintain::Procedure
+    metadata do
+      description 'Disable Puppet feature'
+      confine do
+        feature(:puppet_server)
+      end
+    end
+
+    def run
+      execute!('foreman-rake db:migrate VERSION=0 SCOPE=foreman_puppet') if server_with_puppet_plugin?
+      feature(:installer).run(installer_arguments_disabling_puppet.join(' '), :interactive => false)
+      packages_action(:remove, packages_to_remove, :assumeyes => true)
+    end
+
+    private
+
+    def server_with_puppet_plugin?
+      find_package(foreman_plugin_name('foreman_puppet'))
+    end
+
+    def installer_arguments_disabling_puppet
+      options = []
+      options << '--no-enable-puppet'
+      options << '--puppet-server false'
```

Comment:
This is implied by `--no-enable-puppet`. That it now works is a bug (https://projects.theforeman.org/issues/33821)